### PR TITLE
feat(home): renamed it to home, added test cases, lint config

### PR DESCRIPTION
**Pull request by AlvisonHunterArnuero**
**Hello Team:**

This Pull Request addresses the following updates and improvements:
- **Project Configuration:**
- [x] Added .vscode folder for custom project setup.
- [x] Included `settings.json` & `recommendations.json` files for consistent developer environment settings.
- [x] Created a custom `commitlint` configuration file to enforce commit message standards.

- **Code Enhancements:**
- [x] Renames `Hello` component to `Home` Component  for better clarity and alignment with its purpose..
- [x] Expanded test coverage in both the `Home `and `MyButton` test files.
- [x] Updated TypeScript interfaces to reflect the latest project requirements.

- **Testing:**
Verified updates with successful runs on the following test suites:
- [x] [Home.test.tsx](https://github.com/Krysthecoder/magicooker/blob/855e7c9fdc8b42ac331f46fa4f4a7b795894599a/src/components/Home.test.tsx)
- [x] [MyButton.test.tsx](https://github.com/Krysthecoder/magicooker/blob/855e7c9fdc8b42ac331f46fa4f4a7b795894599a/src/components/MyButton.test.tsx)

Please review the changes at your convenience. Your feedback would be greatly appreciated!

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-import Hello from './components/Hello';
+import Home from './components/Home';
 
 function App() {
   return (
     <>
-      <Hello title="Holiwis, como estan" />
+      <Home title="Holiwis, como estan" />
     </>
   );
```
Comment (Ralexs0096) [inline src/App.tsx]: We can remove the `fragment` since we are only returning one element.

File: src/components/Home.test.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { render, screen } from '@testing-library/react';
+import Home from './Home';
+
+describe('Hello Component', () => {
```
Comment (Ralexs0096) [inline src/components/Home.test.tsx]:
```suggestion
describe('Home Component', () => {
```

File: src/components/Home.test.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  it('renders with title << Hello world >>', () => {
+    render(<Home title="Hello world" />);
+    const testText = screen.getByText('Hello world');
+    expect(testText).toBeInTheDocument();
```
Comment (Ralexs0096) [inline src/components/Home.test.tsx]:
```suggestion
    expect(
      screen.getByText('Hello world')
    ).toBeInTheDocument();
```

File: src/components/types/index.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
...
   onClick: React.MouseEventHandler<HTMLButtonElement>;
   children: React.ReactNode;
   disabled?: boolean;
 }
 
-export interface IHelloProps {
-  title: string;
-}
\ No newline at end of file
+export interface HomeProps {
```
Comment (Ralexs0096) [inline src/components/types/index.ts]: @Krysthecoder @Cema2019 ❗ important: Using I before our interfaces is not a Typescript convention but a C# one. So let's try to keep this as a convention for this and future projects.

**Comment by Ralexs0096** [comment]
@AlvisonHunterArnuero  we already have a develop branch, it would be great to change the target of this branch to develop.

**Comment by AlvisonHunterArnuero** [comment]
> @AlvisonHunterArnuero we already have a develop branch, it would be great to change the target of this branch to develop.

Oh, I didn't know we had a develop repo, I could just switch it to that one if needed. We need to establish the git-flow that will be used on this project. My changes are merely to prepare the init boilerplate for this project, nothing more than that!

**Pull request merged**